src/core/structure/sheet.ts

```diff
@@ -56,24 +56,43 @@ export default class Sheet {
     return this.columns.get(colKey)?.position;
   }
 
-  setCellAt(colPos: number, rowPos: number, value: string): CellInfo {
+  /**
+   * Set the value of a cell at a given position.
+   * If allowEmpty is true, passing an empty value will create the cell.
+   * Only use this parameter when the cell has an incoming reference to avoid creating unused empty cells.
+   */
+  setCellAt(
+    colPos: number,
+    rowPos: number,
+    value: string,
+    allowEmpty: boolean = false,
+  ): CellInfo {
     const position = this.initializePosition(colPos, rowPos);
-    return this.setCell(position.columnKey!, position.rowKey!, value);
+    return this.setCell(
+      position.columnKey!,
+      position.rowKey!,
+      value,
+      allowEmpty,
+    );
   }
 
-  setCell(colKey: ColumnKey, rowKey: RowKey, formula: string): CellInfo {
+  setCell(
```

Comment:
Also, but why does `setCell` has the potential to return an object with undefined values. 

It is a bit related to my previous comment, but I think it should always return a cell value. 

Even in the current logic, it is very bad. It should just return null if the cell is being deleted or doesn't exist, not an object with undefined values.

Reply:
> Even in the current logic. It should just return null if the cell is being deleted or doesn't exist, not an object with undefined values.

It is useful to be able to get the column and row keys for the cell when the `setCell` call is used to delete a cell. I don't see a reason to omit this information.

> Also, I think the fields on CellInfo should not be nullable. And the absence of the cell should be expressed with a null CellInfo not an object with undefined fields.

Calling `setCell` can lead to the cell not existing; I don't think we can get around that. Even if we remove the `deleteCell` call, we should still be cleaning up with `deleteCellIfUnused`, which can lead to the target cell being deleted.

Reply:
If someone passes a none existing colKey and rowKey, the function will return:
`{value: undefined, state: undefined, position:{ rowKey: undefined, columnKey: undefined }}`
I think it is not a good practice to return such an object.

Reply:
True, in that case we should be returning null of course.